src/WriteTask.ts

```diff
@@ -19,9 +19,20 @@ export class WriteTask extends ExifToolTask<void> {
     optionalArgs: string[] = []
   ): WriteTask {
     const sourceFile = _path.resolve(filename)
-    const args = Object.keys(tags)
+
+    const args: Array<string> = []
+
+    Object.keys(tags)
       .filter(k => typeof k === "string" && tags.propertyIsEnumerable(k))
-      .map((key: keyof Tags) => `-${key}=${tags[key]}`)
+      .forEach((key: keyof Tags) => {
+        if (Array.isArray(tags[key])) {
+          const array = tags[key] as Array<string | number>
```

Comment:
Ah, because the value wasn't `const`ed, and could have changed between the two calls. I extracted a `value` const and TypeScript behaves as I hoped it would now.